### PR TITLE
puppetlabs-apt 2.0 support

### PR DESCRIPTION
**Pull request by taik0**
Minor changes to add apt 2.0 module support.
The required_packages param does not exists anymore in apt 2.0, we ensure that apt-transport-https package is present before the apt::source using resource ordering.

### CODE REVIEW

File: manifests/repo/debian.pp [hunk 1]
```diff
@@ -26,11 +26,12 @@
       release           => $release,
       repos             => $repos,
       pin               => $pin,
-      include_src       => false,
-      required_packages => ['apt-transport-https'],
+      include           => { 'src' => false },
```
Comment (pschichtel) [inline manifests/repo/debian.pp]: should this module even decide if I want sources?
Removing the option would make it compatible with apt module versions older than 2.0.0 as well.

**Comment by pschichtel** [comment]
any status on this?

**Comment by joschi** [comment]
@pschichtel Please be patient. We'll review and test the change when we've got time.

**Comment by bernd** [comment]
We started a new puppet module over at https://forge.puppet.com/graylog/graylog for Graylog 2.0.
This module only works for Graylog 1.x and will be deprecated soon.

Is apt 2.0 support actually needed in the old repository? @taik0 @pschichtel

**Comment by pschichtel** [comment]
I don't need it then

**Comment by taik0** [comment]
Because a lot of modules already use apt 2.0, you can't install this module if you have any other module using apt 2.0:
```
root@test:/etc/puppet# puppet module install graylog2-graylog2
Notice: Preparing to install into /etc/puppet/modules ...
Notice: Downloading from http://forge.bcn.sys.devel ...
Error: Could not install module 'graylog2-graylog2' (???)
  No version of 'graylog2-graylog2' can satisfy all dependencies
    Use `puppet module install --ignore-dependencies` to install only this module
``` 

I just wanted to be able to install this module.
You asked for a PR, wasn't planned a new module 2 week ago?
It's not gonna be merged?

**Comment by joschi** [comment]
@taik0 The new Puppet module (https://github.com/Graylog2/puppet-graylog) will not be merged into this repository.

**Comment by taik0** [comment]
@joschi I was talking about this PR. But I will fork this module and apply this  PR to make it work (something I should have done since the beginning).

**Comment by kpacha** [comment]
@joschi I don't get it... 

1- the issue #21 has been open since april 2015. It's a well known problem you face using the module.
2- when people started to ask for a fix, you ask them for a PR (fair enought)
3- now @bernd and you just discard the collaboration cause there is a 'new puppet module'??? seriously???

I guess **people won't feel too willing to collaborate with your projects** if this is the way you handle it.

Bottom line: people like me will start using forks rather than going with this 'official' module

**Comment by bernd** [comment]
@kpacha I understand that this not really optimal. I started the second module because the maintenance burden for this module was pretty high and we currently do not have the resources to maintain it. If you or @taik0 want to help maintaining the this old module for Graylog 1.x, I will happily give you commit bit. :smiley:
